hkmc2/shared/src/main/scala/hkmc2/semantics/Resolver.scala

```diff
@@ -467,18 +467,28 @@ class Resolver(tl: TraceLogger)
     *    the semantic of the term, so it has to done before the symbol
     *    resolution.
     *
+    * @param inCtxPrefix if true, the currently resolving term is the
+    * prefix of an App where the implicit arguments are explicitly
+    * specified, e.g., `f(using 42)`. The implicit arguments should be
+    * resolved on the the App `f(using 42)`, but not on the base of the
+    * App `f`.
     * @param inTyPrefix if true, the currently resolving term is the
-    * prefix of an TyApp which the implicit arguments shouldn't be
-    * resolved on it, but on the TyApp instead.
+    * prefix of an TyApp, e.g., `f[Int]`. The implicit arguments should
+    * be resolved on the the TyApp `f[Int]`, but not on the base of the
+    * TyApp `f`.
     */
-  def resolve(t: Resolvable, inTyPrefix: Bool = false)(using ICtx): (Opt[CallableDefinition], ICtx) =
+  def resolve(t: Resolvable, inCtxPrefix: Bool = false, inTyPrefix: Bool = false)(using ICtx): (Opt[CallableDefinition], ICtx) =
```

Comment:
I can't find anywhere `inTyPrefix` is passed as `true`. If it is, the call site is missing `inTyPrefix =`.

I used to think this pattern was one of the only places where default arguments are fine in internal code, but even this usage feels to error-prone to me, now.

It would be better to remove the defaul arguments and instead use different functions, like `resolveOther`, `resolveCtxPrefix`, and `resolveTyPrefix`, which pass the correct arguments to `resolve`.

Reply:
No wait, I think it couldn't be refactored like that because there are places inheriting the `inPrefix` state to the next recursive call. Instead I'll just remove the default parameters